File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/comment/direct/Java/C++/s826491904_commented.cpp
#include <iostream>
#include <algorithm>

int main() {
    int N;
    std::cin >> N;
    int max_val, min_val;
    std::cin >> max_val;
    min_val = max_val;
    for (int i = 1; i < N; ++i) {
        int num;
        std::cin >> num;
        max_val = std::max(max_val, num);
        min_val = std::min(min_val, num);
    }
    std::cout << max_val - min_val << std::endl;
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/comment/direct/Java/Python/s826491904_commented.py
import sys

def main():
    data = list(map(int, sys.stdin.read().split()))
    N = data[0]
    max_val = data[1]
    min_val = max_val
    for i in range(2, N+1):
        num = data[i]
        max_val = max(max_val, num)
        min_val = min(min_val, num)
    print(max_val - min_val)

if __name__ == "__main__":
    main()
# End of Code
